src/library.js

```diff
@@ -2889,14 +2894,21 @@ LibraryManager.library = {
     buf >>= 2;
     while (ch = HEAPU8[sigPtr++]) {
 #if ASSERTIONS
-      assert(ch === 100/*'d'*/ || ch === 102/*'f'*/ || ch === 105 /*'i'*/ || ch === 106 /*'j'*/, 'Invalid character ' + ch + '("' + String.fromCharCode(ch) + '") in readAsmConstArgs! Use only "d", "f" or "i", and do not specify "v" for void return argument.');
+      const chr = String.fromCharCode(ch);
+      assert(['p', 'd', 'f', 'i', 'j'].includes(chr), 'Invalid character ' + ch + '("' + chr + '") in readAsmConstArgs! Use only "d", "f", "p", "j" or "i", and do not specify "v" for void return argument.');
 #if !WASM_BIGINT
-      assert(ch !== 106/*'j'*/, "i64 arguments to ASM_JS function are not available without WASM_BIGINT");
+      assert(chr !== 'j', "i64 arguments to ASM_JS function are not available without WASM_BIGINT");
 #endif
 #endif
       // Floats are always passed as doubles, and doubles and int64s take up 8
       // bytes (two 32-bit slots) in memory, align reads to these:
-      buf += (ch != 105) & buf;
+      buf += (ch != 105/*i*/) & buf;
```

Comment:
```suggestion
      buf += (chr != 'i') & buf;
```

Reply:
(or was this intentional for speed?)

Reply:
Yeah the `chr = String.fromCharCode(ch);` is only done in ASSERTIONS.  The default codepath still uses comparison with integer constant.   Saves a bit on code size I guess.